types/tx/middleware.go

```diff
@@ -21,12 +22,33 @@ type ResponseSimulateTx struct {
 	Result  *sdk.Result
 }
 
+type Response struct {
+	GasWanted    uint64
+	GasUsed      uint64
+	MsgResponses []*codectypes.Any // Represents each Msg service handler's response type. Will get proto-serialized into the `Data` field in ABCI, see note #2
```

Comment:
```suggestion
	// MsgResponses is an array containing each Msg service handler's response
	// type, packed in an Any. This will get proto-serialized into the `Data` field
	// in the ABCI Check/DeliverTx responses.
	MsgResponses []*codectypes.Any
```